### ISSUE TITLE
FR: Scroll refresh rate not being locked to 30 Hz

### ISSUE THREAD

**Issue by Knurek-NC** [koreader/koreader#12947]
Would it be possible to up the scroll refresh rate for continuous mode up to system's reported refresh rate. It's not an issue for eink screens, obviously, but for LCD devices, especially ones with refresh rates higher than 60, it's immediately noticable the scroll is choppy.

I've tried playing with the e-ink options, they don't seem to help in any way.
Tested on a Samsung S9 Tablet, which has a 120 Hz refresh rate.

**Comment by Frenzie** [comment]
Due to the way all that is implemented (i.e., ereader first) it's unlikely to work nicely.

It looks like it can't currently be easily user patched, but maybe only doing rate in ReaderPaging would do the trick.
https://github.com/koreader/koreader/blob/7151f3cd07356e3cd99144b5f4e783af328df723/frontend/apps/reader/readerui.lua#L114

**Comment by pazos** [comment]
Even 30hz should be plenty if we could take advantage of specific platform's toolkits. Sadly we cannot. See https://github.com/koreader/koreader/issues/12844#issuecomment-2518253732

**Comment by Frenzie** [comment]
It's still smoother at more than 30 with the way it works provided the CPU can keep up. It depends on what's meant by "choppy" of course, if it's something like loading the next page it won't be currently.

**Comment by pazos** [comment]
I think "choppy" means "not a polished Android experience", and that cannot be fixed bumping up numbers :/

**Comment by Frenzie** [comment]
Fair, but either way I think the current values should probably be more easily overridable. I'm more concerned about modern faster eink devices (where it could be part of the device definition) than about 120 Hz devices but the underlying issue is identical. I doubt I'll actively do much about it in the near future though. :-)

**Comment by Euphoriyy** [comment]
I made a patch to change the pan rate [here](https://github.com/Euphoriyy/KOReader.patches#-2-custom-pan-ratelua). It defaults to the screen's refresh rate on Desktop/Linux, where I set it to 144 Hz which works nicely for me.

I am going to test it on my 90 Hz phone later, but I think this feature is definitely pretty useful at least for me. Though it seems to be more noticeable with inertial scrolling off. I could make a PR for this if it's welcome. (I moved the menu to the "Gesture intervals" menu.)

EDIT: This should be expected but after testing it on my phone, it didn't have the same effect and was just as, if not less, smooth than the standard rate of 30 Hz. 

https://github.com/user-attachments/assets/c0c46c4e-c91b-4bc7-8491-c7db30cad325

**Comment by Frenzie** [comment]
> I could make a PR for this if it's welcome.

Yes please. A PR would need to be structured slightly differently but that's probably fairly self-explanatory.
